core/src/ldml/ldml_processor.cpp

```diff
@@ -331,25 +337,23 @@ size_t ldml_processor::process_output(km_core_state *state, const std::u32string
   /** The new context to be added */
   std::u32string new_ctxtstr_changed(ctxt_prefix.second,ctxtstr_cleanedup.end());
 
-  // drop the old suffix. Note: this mutates old_ctxtstr_changed.
+  // FIRST drop the old suffix. Note: this mutates old_ctxtstr_changed.
+  // see remove_text() docs, this PUSHes actions, POPs context items, and TRIMS the string.
   remove_text(state, old_ctxtstr_changed, old_ctxtstr_changed.length());
   assert(old_ctxtstr_changed.length() == 0);
   // old_ctxtstr_changed is now empty because it's been removed.
   // context is "aa" in the above example.
-  emit_text(state, new_ctxtstr_changed);
 
-  // TODO-LDML: need to emit marker here - need to emit text w/ markers, and handle appropriately.
-  // // TODO-LDML: 1-marker hack! need to support a string with intermixed markers.
-  if (str.length() == 3 && str[0] == LDML_UC_SENTINEL && str[1] == LDML_MARKER_CODE) {
-    emit_marker(state, str[2]);
-  }
```

Comment:
@mcdurdin this is the hack I was mentioning. This PR is thus now a major boost to #9468